packages/utils/src/requestdata.ts

```diff
@@ -0,0 +1,417 @@
+/* eslint-disable complexity */
+/**
+ * The functions here, which enrich an event with request data, are mostly for use in Node, but are safe for use in a
+ * browser context. They live here in `@sentry/utils` rather than in `@sentry/node` so that they can be used in
+ * frameworks (like nextjs), which, because of SSR, run the same code in both Node and browser contexts.
+ *
+ * TODO (v8 / #5257): Remove the note below
+ * Note that for now, the tests for this code have to live in `@sentry/node`, since they test both these functions and
+ * the backwards-compatibility-preserving wrappers which still live in `handlers.ts` there.
```

Comment:
Let's make sure we update https://github.com/getsentry/sentry-javascript/issues/5194 with a link to this PR than.

Kind of annoying to dedup, but I think it'll be worth it in the end.

Reply:
Was on my list, but good reminder. Done.